### PR TITLE
docs: Add API list to be ported.

### PR DESCRIPTION
**Pull request by thapav**
Our porting guide does not provide entire api list to be ported
This document provides the api list, references, file location
and header file which has the API prototypes.
A link to the document has also been added to the TizenRT PortingGuide.

Signed-off-by: Vidisha <thapa.v@samsung.com>

### CODE REVIEW

File: docs/APIListToBePorted.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   Valid MTD device structure reference on success; a NULL on failure
+```
+*File Location:*	"os/arch/arm/src/chip/chip_flash.c"
+
+*Prototype Location:*	"os/include/tinyara/fs/mtd.h"
+
+- Add low level chip specific API **up_idle()** to be executed when their is no other ready-to-run task. This is processor idle time and will continue until some interrupt occurs to cause a context switch from the idle task. Processing in this state may be processor-specific. e.g., this is where power management operations might be performed.
```
Comment (jeongchanKim) [inline docs/APIListToBePorted.md:39]: How about changing the line start from `e.g. this is where ~~~`?

File: docs/APIListToBePorted.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+a. **up_putc()** : Output one byte on the serial console
+```c
+ * Prototype: int up_putc(int ch)
+ * Input Parameters:
+ *   ch - chatacter to output
+ * Returned Value:
+ *   sent character
+```
+b. **up_getc()** : Read one byte from the serial console
+```
```
Comment (jeongchanKim) [inline docs/APIListToBePorted.md:115]: We'd better marking this box as c language like line 107.

File: docs/APIListToBePorted.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+ * Prototype: int up_getc(void)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   int value, -1 if error, 0~255 if byte successfully read
+```
+c. **up_puts()** : Output string on the serial console
+```
+ * Prototype: void up_puts(const char *str)
```
Comment (jeongchanKim) [inline docs/APIListToBePorted.md:124]: Same as above.

Comment (thapav) [reply docs/APIListToBePorted.md:124]: done

File: docs/APIListToBePorted.md (hunk 2 above)
Comment (thapav) [reply docs/APIListToBePorted.md]: done

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**